packages/react-components/react-tabster/src/hooks/useFocusVisible.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { Provider_unstable } from '@fluentui/react-shared-contexts';
+import { renderHook } from '@testing-library/react-hooks';
+import * as React from 'react';
+
+jest.mock('../focus/focusVisiblePolyfill', () => ({ applyFocusVisiblePolyfill: jest.fn() }));
+
+import { applyFocusVisiblePolyfill } from '../focus/focusVisiblePolyfill';
+import { useFocusVisible } from './useFocusVisible';
+
+describe('useFocusVisible', () => {
+  describe('targetWindow', () => {
+    it('uses a window from context by default', () => {
+      const Wrapper: React.FC<{ targetDocument: Document | undefined }> = props => (
+        <Provider_unstable value={{ dir: 'ltr', targetDocument: props.targetDocument }}>
+          {props.children}
+        </Provider_unstable>
+      );
+      const element = document.createElement('div');
+
+      const { result, rerender } = renderHook<
+        { targetDocument: Document | undefined },
+        React.MutableRefObject<HTMLElement | null>
+      >(() => useFocusVisible(), {
+        wrapper: Wrapper,
+        initialProps: { targetDocument: undefined },
+      });
+
+      result.current.current = element;
+      rerender({ targetDocument: document });
```

Comment:
I think we should use `createDocument` and use it in both of these tests, it should be safer not to rely on the global document here since we want to be **sure** that we know where the document is coming from

Reply:
@ling1726 please check 59cc184

Reply:
Looks good, thx